src/components/FeatureViews/Accounts/TabComponents/AdminProfileSettings.js

```diff
@@ -182,41 +183,45 @@ function AdminProfileSettings({ user }) {
         setGoogleLoginPromptOpen(false);
     }
 
-    const onFailure = () => {};
+    const onFailure = () => {
+        setGoogleErrorPromptOpen(true);
+    };
 
     const onSuccess = (response) => {
         setGoogleLoginPromptOpen(false);
-
-        setGClassSetting(!gClassSetting);
-        setAdminGCEnabled({
-            variables: {
-                userID: userInfo.user.id,
-                adminType: userInfo.adminType,
-                googleAuthEnabled: !gClassSetting,
-            },
-        });
-        refreshTokenSetup(response).then(() => {
-            getCourses();
-        });
-        setGoogleAuthEmail({
-            variables: {
-                userID: userInfo.user.id,
-                adminType: userInfo.adminType,
-                googleAuthEmail: response.profileObj.email,
-            },
-        });
+        // Check that Google responded with a valid email
+        if (isEmail(response.profileObj.email)) {
+            setGClassSetting(true);
+            refreshTokenSetup(response).then(() => {
```

Comment:
what does "refreshTokenSetup" do? is the idea that if you successfully integrate then you get the courses?

Reply:
Michael found this, it was something to do with making sure we are fetching the newest google classroom token